src/transformers/tokenization_utils_base.py

```diff
@@ -2663,9 +2663,15 @@ def pad(
         # If we have PyTorch/TF/NumPy tensors/arrays as inputs, we cast them as python objects
         # and rebuild them afterwards if no return_tensors is specified
         # Note that we lose the specific device the tensor may be on for PyTorch
+
         first_element = encoded_inputs["input_ids"][0]
-        if isinstance(first_element, (list, tuple)) and first_element:
-            first_element = first_element[0]
+        if isinstance(first_element, (list, tuple)):
+            # first_element might be an empty list/tuple in some edge cases so we grab the first non empty element.
+            index = 0
+            while len(encoded_inputs["input_ids"][index]) == 0:
+                index += 1
+            if index < len(encoded_inputs["input_ids"]):
+                first_element = encoded_inputs["input_ids"][index][0]
```

Comment:
This will still error out if `encoded_inputs["input_ids"]` is fully empty (while in theory in this case we don't really need to error out but can just return a no-op).

Maybe update the following line (`if not isinstance(first_element, int):`) to be `if not isinstance(first_element, (list, tuple, int)):`?

Reply:
Oh, clever! Will do with a comment on why.